src/audio/kpb.c

```diff
@@ -937,6 +939,24 @@ static bool kpb_has_enough_history_data(struct comp_data *kpb,
 	return buffered_data >= his_req;
 }
 
+static inline bool kpb_is_sample_width_supported(uint32_t sampling_width)
+{
+	bool ret;
+
+	switch (sampling_width) {
+	case 16:
```

Comment:
just go with:

```
> 
> case 16:
> case 24:
>    ret = true;
>    break;
> 
```

Reply:
If you are going to do that, please add also `/* FALLTHRU */` Like there: https://github.com/thesofproject/sof/blob/291525d271f3df30ec5134db914a46dec752d1c3/rimage/manifest.c#L179

@lgirdwood 
Why don't we have `-Wimplicit-fallthrough` for FW, when we already have it for Rimage?
It was not active in autotools, but is it just because you are worried about pre-gcc7 support?
I can make it enabled only if compiler supports it (unlike rimage that has it enabled always).

Reply:
+1

Reply:
Updated.

Reply:
@jajanusz no reason why we should not have fall through in CC args.